src/wp-includes/https-migration.php

```diff
@@ -0,0 +1,141 @@
+
+	/**
+	 * Filters whether WordPress should replace old HTTP URLs to the site with their HTTPS counterpart.
+	 *
+	 * If a WordPress site had its URL changed from HTTP to HTTPS, by default this will return `true`. This filter can
+	 * be used to disable that behavior, e.g. after having replaced URLs manually in the database.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param bool $should_replace_insecure_home_url Whether insecure HTTP URLs to the site should be replaced.
+	 */
+	return apply_filters( 'wp_should_replace_insecure_home_url', $should_replace_insecure_home_url );
+}
+
+/**
+ * Replaces insecure HTTP URLs to the site in the given content, if configured to do so.
+ *
+ * This function replaces all occurrences of the HTTP version of the site's URL with its HTTPS counterpart, if
+ * determined via {@see wp_should_replace_insecure_home_url()}.
+ *
+ * @since 5.7.0
+ *
+ * @param string $content Content to replace URLs in.
+ * @return string Filtered content.
+ */
+function wp_replace_insecure_home_url( $content ) {
+	if ( ! wp_should_replace_insecure_home_url() ) {
+		return $content;
+	}
+
+	$https_url = home_url( '', 'https' );
+	$http_url  = str_replace( 'https://', 'http://', $https_url );
+
+	// Also replace potentially escaped URL.
+	$escaped_https_url = str_replace( '/', '\/', $https_url );
+	$escaped_http_url  = str_replace( '/', '\/', $http_url );
+
+	return str_replace(
+		array(
+			$http_url,
+			$escaped_http_url,
+		),
+		array(
+			$https_url,
+			$escaped_https_url,
+		),
+		$content
+	);
+}
+
+/**
+ * Update the 'home' and 'siteurl' option to use the HTTPS variant of their URL.
+ *
+ * If this update does not result in WordPress recognizing that the site is now using HTTPS (e.g. due to constants
+ * overriding the URLs used), the changes will be reverted. In such a case the function will return false.
+ *
+ * @since 5.7.0
+ *
+ * @return bool True on success, false on failure.
+ */
+function wp_update_urls_to_https() {
+	// Get current URL options.
+	$orig_home    = get_option( 'home' );
+	$orig_siteurl = get_option( 'siteurl' );
+
+	// Get current URL options, replacing HTTP with HTTPS.
+	$home    = str_replace( 'http://', 'https://', $orig_home );
+	$siteurl = str_replace( 'http://', 'https://', $orig_siteurl );
+
+	// Update the options.
+	update_option( 'home', $home );
+	update_option( 'siteurl', $siteurl );
+
+	if ( ! wp_is_using_https() ) {
+		// If this did not result in the site recognizing HTTPS as being used,
+		// revert the change and return false.
+		update_option( 'home', $orig_home );
+		update_option( 'siteurl', $orig_siteurl );
+		return false;
+	}
+
+	// Otherwise the URLs were successfully changed to use HTTPS.
+	return true;
+}
+
+/**
+ * Updates the 'https_migrated' option if needed when the given URL has been updated from HTTP to HTTPS.
+ *
+ * If this is a fresh site, a migration will not be necessary, so the migration will be flagged as '1'. Otherwise, it
+ * will be flagged as '0'.
```

Comment:
This kind of feels backwards to me. I think mainly because, AFAICT, this "migration" is never completed. It isn't like we set it to `0`, run a process to update all the URLS in the db, and then set it to `1`. It is only set to `1` if this is performed on a completely new site.

Based on the core usage, I would've expected something like `https_migration_required = 1`.